build.gradle

```diff
@@ -95,32 +98,69 @@ task generateJniInterfaces(type:Exec) {
-    // update is made on classes.
-    outputs.upToDateWhen {
-        def smallestGeneratedTimestamp = targets.collect {
-            new File(project.rootDir, "wasmtime-jni/src/${it.replace(".", "_")}/mod.rs").lastModified()
-        }.min()
-        def largestSourceTimestamp = targets.collect {
-            def path = "classes/java/main/" + it.replace(".", "/") + ".class"
-            new File(project.buildDir, path).lastModified()
-        }.max()
-        smallestGeneratedTimestamp >= largestSourceTimestamp
+    if(jniLibOsClassifier().equals("windows")){
+        File file = new File("./generateJniInterfacesBat.bat")
+        file.createNewFile()
+        file.deleteOnExit()
+        String cmd = ""
+        for(String each:(["cmd","java -cp ${project.buildDir}\\tmp;" +
+                "${project.buildDir}\\tmp\\javah4x;" +
+                "${project.buildDir}\\tmp\\javah4x\\rust;" +
+                "${project.buildDir}\\classes\\java\\main\\io\\github\\kawamuray\\wasmtime;" +
+                "${project.buildDir}\\classes\\java\\main\\io\\github\\kawamuray\\wasmtime\\wasi" +
+                " javah4x.Javah4x", 'rust', '.\\wasmtime-jni\\src'] + targets)){
+            cmd+=(each+" ")
+        }
+        file.write(cmd)
+        commandLine("generateJniInterfacesBat.bat")
+        // Manually compare the source (class files) and output (rust sources) to run this task only when any
+        // update is made on classes.
+        outputs.upToDateWhen {
+            def smallestGeneratedTimestamp = targets.collect {
+                new File(project.rootDir, "wasmtime-jni/src/${it.replace(".", "_")}/mod.rs").lastModified()
+            }.min()
+            def largestSourceTimestamp = targets.collect {
+                def path = "classes/java/main/" + it.replace(".", "/") + ".class"
+                new File(project.buildDir, path).lastModified()
+            }.max()
+            smallestGeneratedTimestamp >= largestSourceTimestamp
+        }
+    }else{
+        commandLine(["${project.buildDir}/tmp/javah4x-${javah4xVersion}", 'rust', './wasmtime-jni/src'] + targets)
+        // Manually compare the source (class files) and output (rust sources) to run this task only when any
+        // update is made on classes.
+        outputs.upToDateWhen {
+            def smallestGeneratedTimestamp = targets.collect {
+                new File(project.rootDir, "wasmtime-jni/src/${it.replace(".", "_")}/mod.rs").lastModified()
+            }.min()
+            def largestSourceTimestamp = targets.collect {
+                def path = "classes/java/main/" + it.replace(".", "/") + ".class"
+                new File(project.buildDir, path).lastModified()
+            }.max()
+            smallestGeneratedTimestamp >= largestSourceTimestamp
+        }
     }
 }
 
 task buildJniLib(type:Exec) {
     dependsOn 'generateJniInterfaces'
-    workingDir './wasmtime-jni'
+    if(jniLibOsClassifier().equals("windows")){
+        workingDir '.\\wasmtime-jni'
```

Comment:
Just like the other fixes can we just keep slash separated paths for windows too?

Reply:
OK.